src/player/Player.php

```diff
@@ -1213,9 +1213,14 @@ protected function processMostRecentMovements() : void{
 
 			$horizontalDistanceTravelled = sqrt((($from->x - $to->x) ** 2) + (($from->z - $to->z) ** 2));
 			if($horizontalDistanceTravelled > 0){
-				//TODO: check swimming (adds 0.015 exhaustion in MCPE)
 				if($this->isSprinting()){
 					$this->hungerManager->exhaust(0.1 * $horizontalDistanceTravelled, PlayerExhaustEvent::CAUSE_SPRINTING);
+				}elseif($this->isSwimming()){
```

Comment:
This won't be set if the player is just bobbing up and down on the surface of water.